curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md

```diff
@@ -36,10 +36,17 @@ assert(
 The link's text should be `cat photos`. You have either omitted the text or have a typo.
 
 ```js
-const nestedAnchor = document.querySelector(`p > a`);
-assert(
-  nestedAnchor.innerText.toLowerCase().replace(/\s+/g, ' ') === 'cat photos'
-);
+const nestedAnchor = document.querySelector('p > a');
+const innerContent = nestedAnchor.innerHTML;
+assert.isTrue(innerContent.trim() === 'cat photos');
+```
+
+The text inside your anchor element has extra leading or trailing whitespace. The only space in the anchor text should be between the word `cat` and the word `photos`.
+
+```js
+const nestedAnchor = document.querySelector('p > a');
+const innerContent = nestedAnchor.innerHTML;
+assert.isNotTrue(/^\s+|\s+$/.test(innerContent));
```

Comment:
Wouldn't this be simpler?

```suggestion
assert(innerContent === 'cat photos'));
```

Reply:
The point is to make it clear what is being tested for and not just rely on the assert message.

Reply:
I don't see how the regex is clearer than "check that the innerContent is exactly this string"

Reply:
We first test what the string _should be_ and we now explicitly test what it shouldn't be.

```
assert.isTrue(expectedValue);
assert.isNotTrue(incorrectValue);
```

vs

```
assert.isTrue(expectedValue)
assert.isTrue(expectedValue)
```

I wrote it like that because initially when first reading the two tests it did actually confuse me a little and explicitly testing for the expected incorrect value seemed easier to read and understand.

Reply:
You care much more than I do about this. I see this as unnecessarily complicated for the target goal, but I withdraw my suggestion.